README.rst
:**NOTE**: **This github repository is archived. The repository contents and history have moved to** `google-cloud-python`_.

.. _google-cloud-python: https://github.com/googleapis/google-cloud-python/tree/main/packages/google-cloud-bigquery-datatransfer


Python Client for BigQuery Data Transfer API
============================================

|stable| |pypi| |versions|

`BigQuery Data Transfer API`_: allows users to transfer data from partner SaaS applications to Google BigQuery on a scheduled, managed basis.

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-bigquery-datatransfer.svg
   :target: https://pypi.org/project/google-cloud-bigquery-datatransfer/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-bigquery-datatransfer.svg
   :target: https://pypi.org/project/google-cloud-bigquery-datatransfer/
.. _BigQuery Data Transfer API: https://cloud.google.com/bigquery/transfer/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/bigquerydatatransfer/latest
.. _Product Documentation:  https://cloud.google.com/bigquery/transfer/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the BigQuery Data Transfer API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the BigQuery Data Transfer API.:  https://cloud.google.com/bigquery/transfer/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a virtual environment using `venv`_. `venv`_ is a tool that
creates isolated Python environments. These isolated environments can have separate
versions of Python packages, which allows you to isolate one project's dependencies
from the dependencies of other projects.

With `venv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`venv`: https://docs.python.org/3/library/venv.html


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/`_ folder.

.. _samples/: https://github.com/googleapis/python-bigquery-datatransfer/tree/main/samples


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current `active`_ and `maintenance`_ versions of
Python.

Python >= 3.7

.. _active: https://devguide.python.org/devcycle/#in-development-main-branch
.. _maintenance: https://devguide.python.org/devcycle/#maintenance-branches

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an `end-of-life`_
version of Python, we recommend that you update as soon as possible to an actively supported version.

.. _end-of-life: https://devguide.python.org/devcycle/#end-of-life-branches

Mac/Linux
^^^^^^^^^

.. code-block:: console

    python3 -m venv <your-env>
    source <your-env>/bin/activate
    pip install google-cloud-bigquery-datatransfer


Windows
^^^^^^^

.. code-block:: console

    py -m venv <your-env>
    .\<your-env>\Scripts\activate
    pip install google-cloud-bigquery-datatransfer

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for BigQuery Data Transfer API
   to see other available methods on the client.
-  Read the `BigQuery Data Transfer API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _BigQuery Data Transfer API Product documentation:  https://cloud.google.com/bigquery/transfer/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/google-cloud-bigquery-datatransfer/#history

## [3.12.1](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.12.0...v3.12.1) (2023-09-22)


### Documentation

* Minor formatting ([091adac](https://github.com/googleapis/python-bigquery-datatransfer/commit/091adac922eae2d5b35816f6dc07913a594bd169))
* Update transferConfig.name description to indicate that it supports both formats ([091adac](https://github.com/googleapis/python-bigquery-datatransfer/commit/091adac922eae2d5b35816f6dc07913a594bd169))

## [3.12.0](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.11.2...v3.12.0) (2023-08-01)


### Features

* Add EncryptionConfiguration to TransferConfig ([f57343c](https://github.com/googleapis/python-bigquery-datatransfer/commit/f57343c31acd8ab44b4cc427e3c148ba93bb5368))
* Add List type to Data source parameter. ([f57343c](https://github.com/googleapis/python-bigquery-datatransfer/commit/f57343c31acd8ab44b4cc427e3c148ba93bb5368))

## [3.11.2](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.11.1...v3.11.2) (2023-07-04)


### Bug Fixes

* Add async context manager return types ([#470](https://github.com/googleapis/python-bigquery-datatransfer/issues/470)) ([551adeb](https://github.com/googleapis/python-bigquery-datatransfer/commit/551adeb66c823df68daed10bd55ad7a9a075a986))

## [3.11.1](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.11.0...v3.11.1) (2023-03-27)


### Documentation

* Fix formatting of request arg in docstring ([#455](https://github.com/googleapis/python-bigquery-datatransfer/issues/455)) ([40a85d6](https://github.com/googleapis/python-bigquery-datatransfer/commit/40a85d6eb5e40ff77c054c76bbdfe8a5b821e1ef))
* Removes deprecated sample ([#451](https://github.com/googleapis/python-bigquery-datatransfer/issues/451)) ([250acb4](https://github.com/googleapis/python-bigquery-datatransfer/commit/250acb4ec89a33c54174659d33fd7088f22f93db))
* Replace deprecated snippet ([e2f04db](https://github.com/googleapis/python-bigquery-datatransfer/commit/e2f04db7553721610903500133553e67fe717642))

## [3.11.0](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.10.1...v3.11.0) (2023-02-28)


### Features

* Enable "rest" transport in Python for services supporting numeric enums ([#439](https://github.com/googleapis/python-bigquery-datatransfer/issues/439)) ([b77f671](https://github.com/googleapis/python-bigquery-datatransfer/commit/b77f6716a47ba5bda4e50919e37b41fbc7cb3e20))


### Documentation

* Minor comment update ([#445](https://github.com/googleapis/python-bigquery-datatransfer/issues/445)) ([27b3271](https://github.com/googleapis/python-bigquery-datatransfer/commit/27b3271682fae002d29c65e057190c01f4f24ed4))

## [3.10.1](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.10.0...v3.10.1) (2023-01-20)


### Bug Fixes

* Add context manager return types ([a10a470](https://github.com/googleapis/python-bigquery-datatransfer/commit/a10a470dc53947e35202c0b296fe74e519c0abe2))


### Documentation

* Add documentation for enums ([a10a470](https://github.com/googleapis/python-bigquery-datatransfer/commit/a10a470dc53947e35202c0b296fe74e519c0abe2))

## [3.10.0](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.9.0...v3.10.0) (2023-01-10)


### Features

* Add support for python 3.11 ([#421](https://github.com/googleapis/python-bigquery-datatransfer/issues/421)) ([d81fe69](https://github.com/googleapis/python-bigquery-datatransfer/commit/d81fe694862b9663e8ffd407c9bad7af6bc4fc01))

## [3.9.0](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.8.0...v3.9.0) (2023-01-04)


### Features

* Add location methods ([#416](https://github.com/googleapis/python-bigquery-datatransfer/issues/416)) ([6538b29](https://github.com/googleapis/python-bigquery-datatransfer/commit/6538b29601e91d16e6a2095a256b4b18920d73b9))

## [3.8.0](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.7.3...v3.8.0) (2022-12-15)


### Features

* Add support for `google.cloud.bigquery_datatransfer.__version__` ([c333464](https://github.com/googleapis/python-bigquery-datatransfer/commit/c3334647fd103e2f10466f1daecde51a89aae727))
* Add typing to proto.Message based class attributes ([c333464](https://github.com/googleapis/python-bigquery-datatransfer/commit/c3334647fd103e2f10466f1daecde51a89aae727))


### Bug Fixes

* Add dict typing for client_options ([c333464](https://github.com/googleapis/python-bigquery-datatransfer/commit/c3334647fd103e2f10466f1daecde51a89aae727))
* **deps:** Require google-api-core &gt;=1.34.0, >=2.11.0  ([20be9ee](https://github.com/googleapis/python-bigquery-datatransfer/commit/20be9ee41126f3c38ef9a6070e7a6fad5a7712d1))
* Drop usage of pkg_resources ([20be9ee](https://github.com/googleapis/python-bigquery-datatransfer/commit/20be9ee41126f3c38ef9a6070e7a6fad5a7712d1))
* Fix timeout default values ([20be9ee](https://github.com/googleapis/python-bigquery-datatransfer/commit/20be9ee41126f3c38ef9a6070e7a6fad5a7712d1))


### Documentation

* **samples:** Snippetgen handling of repeated enum field ([c333464](https://github.com/googleapis/python-bigquery-datatransfer/commit/c3334647fd103e2f10466f1daecde51a89aae727))
* **samples:** Snippetgen should call await on the operation coroutine before calling result ([20be9ee](https://github.com/googleapis/python-bigquery-datatransfer/commit/20be9ee41126f3c38ef9a6070e7a6fad5a7712d1))

## [3.7.3](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.7.2...v3.7.3) (2022-10-07)


### Bug Fixes

* **deps:** Allow protobuf 3.19.5 ([#400](https://github.com/googleapis/python-bigquery-datatransfer/issues/400)) ([5e71c7c](https://github.com/googleapis/python-bigquery-datatransfer/commit/5e71c7c38ee937a4b61dc6a0f1ba79431a7d97b1))

## [3.7.2](https://github.com/googleapis/python-bigquery-datatransfer/compare/v3.7.1...v3.7.2) (2022-09-29)



owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2023 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.7",
    "3.8",
    "3.9",
    "3.10",
    "3.11",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==22.3.0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.11"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/bigquery_datatransfer_v1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install(
        'mypy',
        'types-requests',
        'types-protobuf'
    )
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
